foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/version_specific/TrinoDeltaLakeConnectorYamlMigration.java

```diff
@@ -0,0 +1,109 @@
+    private static final Logger logger = LoggerFactory.getLogger(TrinoDeltaLakeConnectorYamlMigration.class);
+    private static final String CATALOGS = "catalogs:";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        try {
+            YamlUtils.YamlObject catalogs = getTrinoCatalogs(file);
+            return catalogs!= null && !hasDeltaLakeConnector(catalogs);
+        } catch (YAMLException e) {
+            logger.warn("Failed to parse YAML file, likely because it is a Helm template: {}", file.getName(), e);
+        } catch (IOException e) {
+            throw new BatonException("Failed to read YAML file.", e);
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        try {
+            List<String> updatedContent = new ArrayList<>();
+            List<String> lines = readAllFileLines(file);
+            int indentSpaces = 0;
+            boolean inCatalogs = false;
+            for (int i = 0; i < lines.size(); i++) {
+                String trimmedLine = lines.get(i).trim();
+                if (inCatalogs) {
+                    updatedContent.add(getDeltaLakeConnectorConfig(indentSpaces));
+                    inCatalogs = false;
+                }
+                if(trimmedLine.equals(CATALOGS)) {
+                    indentSpaces = YamlUtils.getIndentSpaces(lines, 0);
+                    inCatalogs = true;
+                }
+                updatedContent.add(lines.get(i));
+            }
+
+            writeFile(file, updatedContent);
+            return true;
+        } catch (IOException e) {
+            throw new BatonException("Failed to migration YAML file.", e);
+        }
+    }
+
+    private boolean hasDeltaLakeConnector(YamlUtils.YamlObject catalogs) {
+        try {
+            String deltaConfig = (String)catalogs.get("delta");
+            return deltaConfig.contains("connector.name=delta_lake");
+        } catch (Exception e) {
+            logger.info("Delta Lake connector not found in trino.catalogs configuration");
+        }
+        return false;
+    }
+
+    private String getDeltaLakeConnectorConfig(int indentSpaces) {
+        return indent(3, indentSpaces) + "delta: |\n" +
+                indent(4, indentSpaces) + "connector.name=delta_lake\n" +
+                indent(4, indentSpaces) + "hive.metastore.uri=thrift://hive-metastore-service:9083\n" +
+                indent(4, indentSpaces) + "fs.native-s3.enabled=true\n" +
+                indent(4, indentSpaces) + "s3.aws-access-key=${ENV:AWS_ACCESS_KEY_ID}\n" +
+                indent(4, indentSpaces) + "s3.aws-secret-key=${ENV:AWS_SECRET_ACCESS_KEY}\n" +
+                indent(4, indentSpaces) + "s3.endpoint=http://s3-local:4566\n" +
+                indent(4, indentSpaces) + "s3.region=us-east-1\n" +
+                indent(4, indentSpaces) + "s3.path-style-access=true\n";
```

Comment:
A: This would make the migration mirror the default file more closely
```suggestion
                indent(4, indentSpaces) + "s3.path-style-access=true";
```